patches/minecraft/net/minecraft/world/entity/Entity.java.patch

```diff
@@ -125,6 +129,33 @@
           this.m_5496_(soundtype.m_56776_(), soundtype.m_56773_() * 0.15F, soundtype.m_56774_());
        }
     }
+@@ -1042,7 +_,7 @@
+       if (this.m_6069_()) {
+          this.m_20282_(this.m_20142_() && this.m_20069_() && !this.m_20159_());
+       } else {
+-         this.m_20282_(this.m_20142_() && this.m_5842_() && !this.m_20159_() && this.f_19853_.m_6425_(this.f_19826_).m_76153_(FluidTags.f_13131_));
++         this.m_20282_(this.m_20142_() && this.m_5842_() && !this.m_20159_() && this.f_19853_.m_6425_(this.f_19826_).m_76153_(FluidTags.f_13131_) || this.f_19853_.m_6425_(this.f_19826_).m_76153_(Tags.Fluids.DEFAULT_FLUID_PHYSIC));
+       }
+ 
+    }
+@@ -1058,7 +_,7 @@
+    void m_20074_() {
+       if (this.m_20202_() instanceof Boat) {
+          this.f_19798_ = false;
+-      } else if (this.m_19943_(FluidTags.f_13131_, 0.014D)) {
++      } else if (this.m_19943_(FluidTags.f_13131_, 0.014D) || this.m_19943_(Tags.Fluids.DEFAULT_FLUID_PHYSIC, 0.014D)) {
+          if (!this.f_19798_ && !this.f_19803_) {
+             this.m_5841_();
+          }
+@@ -1073,7 +_,7 @@
+    }
+ 
+    private void m_20323_() {
+-      this.f_19800_ = this.m_19941_(FluidTags.f_13131_);
++      this.f_19800_ = this.m_19941_(FluidTags.f_13131_) || this.m_19941_(Tags.Fluids.DEFAULT_FLUID_PHYSIC);
```

Comment:
Just as a heads up, you need about 10 more patches to get the minimum necessary to enable proper fluid physics for the player alone. Tag determining physics here is also not that great since it tends to make custom fluid logic difficult to achieve. If you would like to help contribute for this, you can work on making #7992 without any breaking changes since Simon nor I have the time to work on it consistently.

From just looking at the patches, I'm pretty sure this will just keep you in the swimming animation but won't start it in the fluid, do some falling logic without any proper swimming, and desync the eyes in water fluid thus breaking vanilla compatibility.

Reply:
oh ok